File: 1_two_sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const heap: Map<number, number> = new Map();
  const ans: number[] = [];
  nums.forEach((val, idx) => {
    if (heap.has(target - val))
      ans.push(idx, heap.get(target - val) as number);
    else heap.set(val, idx);
  })
  return ans;
};

File: 1_two_sum/solution.cc
#include <vector>
#include <unordered_map>

using std::unordered_map;
using std::vector;
vector<int> twoSum(vector<int> &nums, int target)
{
  unordered_map<int, int> heap;
  for (int i = 0; i < nums.size(); i++)
  {
    int r = target - nums[i];
    if (heap.count(r))
      return {heap[r], i};
    else
      heap[nums[i]] = i;
  }
  return {-1, -1};
}